tests/test_models.py

```diff
@@ -73,7 +73,10 @@ def test_multi_period_default_discount_rate():
     warnings.filterwarnings("ignore", category=FutureWarning)
     timeindex = pd.date_range(start="2017-01-01", periods=100, freq="D")
     es = solph.EnergySystem(
-        timeindex=timeindex, periods={0: timeindex}, infer_last_interval=True
+        timeindex=timeindex,
+        timeincrement=[1] * len(timeindex),
+        periods=[timeindex],
+        infer_last_interval=False,
```

Comment:
I see that you where looking to explicitly set `infer_last_interval`. 

I noticed that this test was somewhat outdated since I used the old `periods` definition (type dict) here which I fixed in #977 which I just merged. 

Another thing is that though it was syntactically correct with the list attribute as in the current dev branch, it would not have worked to set up a model this way. Since the test is only to demonstrate a warning that occurs before the model setup, I'm sorry that have not detected this. Anyways, as it is now, also the model build would work. 

Concerning the (not so nice) inconsistency to be resolved some day, see #979.